src/nlp_expr.jl

```diff
@@ -81,15 +81,11 @@ struct GenericNonlinearExpr{V<:AbstractVariableRef} <: AbstractJuMPScalar
     head::Symbol
     args::Vector{Any}
 
-    function GenericNonlinearExpr(head::Symbol, args::Vector{Any})
-        index = findfirst(Base.Fix2(isa, AbstractJuMPScalar), args)
-        if index === nothing
-            error(
-                "Unable to create a nonlinear expression because it did not " *
-                "contain any JuMP scalars. head = $head, args = $args.",
-            )
-        end
-        return new{variable_ref_type(args[index])}(head, args)
```

Comment:
I understand the motivation for this. Unfortunately, it will break InfiniteOpt, but I can look into refactoring the workflow. If supporting array arguments necessitates this change, then I support it.

Reply:
We can probably just keep this method then. It's not necessary. Let me make some changes.

Reply:
Try this

Reply:
I confirmed that this works with InfiniteOpt: https://github.com/infiniteopt/InfiniteOpt.jl/pull/321

Reply:
Thanks, you beat me to it :)